indexer/database/blocks.go

```diff
@@ -111,14 +108,19 @@ func (db *blocksDB) StoreLegacyStateBatch(stateBatch *LegacyStateBatch) error {
 	return result.Error
 }
 
-func (db *blocksDB) FinalizedL1BlockHeight() (*big.Int, error) {
+// FinalizedL1BlockHeader returns the latest L1 block header stored in the database, nil otherwise
+func (db *blocksDB) FinalizedL1BlockHeader() (*L1BlockHeader, error) {
 	var l1Header L1BlockHeader
 	result := db.gorm.Order("number DESC").Take(&l1Header)
 	if result.Error != nil {
+		if errors.Is(result.Error, gorm.ErrRecordNotFound) {
+			return nil, nil
+		}
```

Comment:
Why is this necessary?
```suggestion
```

Reply:
Since we are returning pointers, I think it makes sense to have `nil` represent an non-existent entry when `err == nil`. Useful for the processor when determining if it should start from genesis or a specified height (last indexed block).

Gorm treats non existent entries as errors